### PR TITLE
[None][perf] Replace Parakeet audio encoder with native trtllm layers

### PR DESCRIPTION
**Pull request by aswinvisva**
Replace HF ParakeetEncoderAttention with ParakeetConformerAttention subclassing trtllm.Attention, and convert all FFN/norm layers to trtllm.Linear/LayerNorm. Adds a logit_bias hook to trtllm.Attention for Transformer-XL RPE injection. 

#### Performance Speedup

| Scenario | Batch | HF mean | TRT mean | Speedup |
| :--- | :--- | :--- | :--- | :--- |
| 10-min audio | 20 clips | 47.0 ms | 45.9 ms | 1.02x |
| 30-min audio | 60 clips | 143.7 ms | 134.8 ms | 1.07x |

#### Numerical Accuracy vs. HF Reference
*(Same weights, same input)*

| Scenario | mean \|diff\| | cos sim median | tokens > 0.99 | tokens > 0.50 |
| :--- | :--- | :--- | :--- | :--- |
| 10-min (B=20) | 0.026 | 0.9957 | 68% | 99.7% |
| 30-min (B=60) | 0.025 | 0.9964 | 71% | 99.8% |

## Summary by CodeRabbit

* **New Features**
  * Attention module now supports optional logit bias parameter for advanced attention masking and control.

* **Improvements**
  * Parakeet encoder implementation optimized with native TRT-LLM Conformer stack for enhanced performance and efficiency.

[Review Change Stack](https://app.coderabbit.ai/change-stack/NVIDIA/TensorRT-LLM/pull/14474?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

## Test Coverage

Ran the layers against the base hugging-face transformers implementation and verified that the outputs were numerically identical.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/_torch/models/modeling_parakeet.py
Comment (2ez4bz) [inline tensorrt_llm/_torch/models/modeling_parakeet.py]: Nit: since we now own this class implementation (instead of having to use HF's), is there a more readable way to define this function? Right now it's a private method on a class that accepts an instance of `ParakeetExtractor`, really only to access a couple of fields in its config. Maybe a free standing function with all the config's parameters properly declared as input arguments?

Comment (2ez4bz) [inline tensorrt_llm/_torch/models/modeling_parakeet.py]: Nice!

Comment (2ez4bz) [inline tensorrt_llm/_torch/models/modeling_parakeet.py]: Just checking: would it make sense to override `forward_impl` instead by passing in `logit_bias`? That way we can leverage the other code in `forward` itself.

File: tensorrt_llm/_torch/modules/attention.py
Comment (brb-nv) [inline tensorrt_llm/_torch/modules/attention.py]: Do we need to assert the uniform length part somewhere?

Comment (brb-nv) [inline tensorrt_llm/_torch/modules/attention.py]: Can we do away with using `vanilla` attention backend? Any reason why need to add this here?
https://github.com/aswinvisva/TensorRT-LLM/blob/f2a884aefddc93837f459e9d2d4b3e9528d12116/tensorrt_llm/_torch/attention_backend/vanilla.py#L251

**Review by brb-nv — COMMENTED** [review]
LGTM.

**Review by moraxu — APPROVED** [review]

**Review by 2ez4bz — APPROVED** [review]
